tests/nodeset-loader/check_memory.sh

```diff
@@ -0,0 +1,435 @@
+#          - Dependent Nodesets:
+#             * $(OPEN62541_NODESET_DIR)Mining/DevelopmentSupport/Dozer/1.0.0/Opc.Ua.Mining.DevelopmentSupport.Dozer.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/DevelopmentSupport/RoofSupportSystem/1.0.0/Opc.Ua.Mining.DevelopmentSupport.RoofSupportSystem.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/Extraction/ShearerLoader/1.0.0/Opc.Ua.Mining.Extraction.ShearerLoader.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/Loading/General/1.0.0/Opc.Ua.Mining.Loading.General.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/Loading/HydraulicExcavator/1.0.0/Opc.Ua.Mining.Loading.HydraulicExcavator.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/MineralProcessing/RockCrusher/1.0.0/Opc.Ua.Mining.MineralProcessing.RockCrusher.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/PELOServices/FaceAlignmentSystem/1.0.0/Opc.Ua.Mining.PELOServices.FaceAlignmentSystem.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/PELOServices/General/1.0.0/Opc.Ua.Mining.PELOServices.General.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/TransportDumping/ArmouredFaceConveyor/1.0.0/Opc.Ua.Mining.TransportDumping.ArmouredFaceConveyor.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/TransportDumping/General/1.0.0/Opc.Ua.Mining.TransportDumping.General.NodeSet2.xml
+#             * $(OPEN62541_NODESET_DIR)Mining/TransportDumping/RearDumpTruck/1.0.0/Opc.Ua.Mining.TransportDumping.RearDumpTruck.NodeSet2.xml
+#      - $(OPEN62541_NODESET_DIR)PADIM/Opc.Ua.IRDI.NodeSet2.xml
+#          - Dependent Nodesets:
+#             * $(OPEN62541_NODESET_DIR)PADIM/Opc.Ua.PADIM.NodeSet2.xml
+#      - $(OPEN62541_NODESET_DIR)POWERLINK/Opc.Ua.POWERLINK.NodeSet2.xml
+#      - $(OPEN62541_NODESET_DIR)Pumps/Opc.Ua.Pumps.NodeSet2.xml
+#      - $(OPEN62541_NODESET_DIR)Scales/Opc.Ua.Scales.NodeSet2.xml
+#
+
+set -e
+IFS=
+
+function print_nodesets() {
+
+    echo "Nodesets:"
+	shift
+
+	for i in $@
+    do
+        echo "      $(basename $i)"
+    done
+}
+
+function total_memory_usage() {
+
+    echo -ne "Total memory usage: "
+
+	TEST_APP=$1
+	shift
+
+	touch tmp.log
+	valgrind --trace-children=yes --tool=memcheck --log-file=tmp.log $TEST_APP $@ &> /dev/null
+	grep -o -P '(?<=frees,\ ).*(?=\ bytes\ allocated)' tmp.log | echo "$(tail -n 1)" "B"
+	rm tmp.log
+}
+
+function memory_usage() {
+    echo -en "|\n-> "
+    print_nodesets $@
+    echo -en "|\n-> "
+    total_memory_usage $@
+    echo; echo
+}
+
+UA_NODESET_PATH=/usr/local/share/open62541/tools/ua-nodeset
+NODESETLOADER_TESTS_PATH=./../../build/bin/tests
+NODESETLOADER_TEST_APP=check_nodeset_loader_input
+
+echo "... ADI ..."
+memory_usage $NODESETLOADER_TESTS_PATH/$NODESETLOADER_TEST_APP $UA_NODESET_PATH/DI/Opc.Ua.Di.NodeSet2.xml \
+							                                   $UA_NODESET_PATH/ADI/Opc.Ua.Adi.NodeSet2.xml
```

Comment:
The memory usage tests are very nice!
One small comments: Use spaces instead of tabs.
This file mixes space and tabs. And I have to scroll right to see the entire text.